pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -89,6 +107,13 @@ func NewPersistentVolumeRecycler(kubeClient client.Interface, syncPeriod time.Du
 				}
 				recycler.reclaimVolume(pv)
 			},
+			DeleteFunc: func(obj interface{}) {
+				pv, ok := obj.(*api.PersistentVolume)
+				if !ok {
+					glog.Errorf("Error casting object to PersistentVolume: %v", obj)
+				}
+				recycler.removeFailedVolume(pv)
```

Comment:
Would need to return if !ok.   This function would fail if pv == DeletedFinalStateUnknown.